### ISSUE TITLE
TransVCL  convert  tensorrt has error，onnx::Concat_2: network input that is shape tensor must have type Int32

### ISSUE THREAD

**Issue by yang1112git** [NVIDIA/TensorRT#3176]
## Environment

**TensorRT Version**: 8.5.3

**NVIDIA GPU**: V100 

**NVIDIA Driver Version**: 470.82.01

**CUDA Version**: 11.4

**CUDNN Version**: 

Operating System: ubuntu 18.04

Python Version (if applicable): 3.8

Tensorflow Version (if applicable):

PyTorch Version (if applicable):  2.0.1

Baremetal or Container (if so, version):  Container 

## Relevant Files

**Model link**: https://github.com/transvcl/TransVCL

## Steps To Reproduce

I would like to convert a Torch model to a TensorRT model. Inference with ONNX works fine, but I encountered an error when converting to TRT. Please help me look into how to resolve it.
torch to onnx is fine，onnx  to tensorrt has error ：
`[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Constant_9_output_0
[08/01/2023-08:17:33] [V] [TRT] /Squeeze [Squeeze] inputs: [/Split_output_0 -> (1, 1200, 1200)[FLOAT]], [/Constant_9_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Original shape: (1, 1200, 1200), squeezing to: (1200, 1200)
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /Squeeze for ONNX node: /Squeeze
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /Squeeze_output_0 for ONNX tensor: /Squeeze_output_0
[08/01/2023-08:17:33] [V] [TRT] /Squeeze [Squeeze] outputs: [/Squeeze_output_0 -> (1200, 1200)[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Gather_2 [Gather]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: onnx::Concat_2
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /pos_encoding/Constant_2_output_0
[08/01/2023-08:17:33] [V] [TRT] /Gather_2 [Gather] inputs: [onnx::Concat_2 -> (1, 1200)[FLOAT]], [/pos_encoding/Constant_2_output_0 -> ()[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Using Gather axis: 0
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /Gather_2 for ONNX node: /Gather_2
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /Gather_2_output_0 for ONNX tensor: /Gather_2_output_0
[08/01/2023-08:17:33] [V] [TRT] /Gather_2 [Gather] outputs: [/Gather_2_output_0 -> (1200)[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /ReduceSum [ReduceSum]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Gather_2_output_0
[08/01/2023-08:17:33] [V] [TRT] /ReduceSum [ReduceSum] inputs: [/Gather_2_output_0 -> (1200)[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /ReduceSum for ONNX node: /ReduceSum
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /ReduceSum_output_0 for ONNX tensor: /ReduceSum_output_0
[08/01/2023-08:17:33] [V] [TRT] /ReduceSum [ReduceSum] outputs: [/ReduceSum_output_0 -> ()[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Cast [Cast]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /ReduceSum_output_0
[08/01/2023-08:17:33] [V] [TRT] /Cast [Cast] inputs: [/ReduceSum_output_0 -> ()[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Casting to type: int32
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /Cast for ONNX node: /Cast
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /Cast_output_0 for ONNX tensor: /Cast_output_0
[08/01/2023-08:17:33] [V] [TRT] /Cast [Cast] outputs: [/Cast_output_0 -> ()[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Gather_3 [Gather]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: onnx::Concat_3
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /pos_encoding/Constant_2_output_0
[08/01/2023-08:17:33] [V] [TRT] /Gather_3 [Gather] inputs: [onnx::Concat_3 -> (1, 1200)[FLOAT]], [/pos_encoding/Constant_2_output_0 -> ()[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Using Gather axis: 0
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /Gather_3 for ONNX node: /Gather_3
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /Gather_3_output_0 for ONNX tensor: /Gather_3_output_0
[08/01/2023-08:17:33] [V] [TRT] /Gather_3 [Gather] outputs: [/Gather_3_output_0 -> (1200)[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /ReduceSum_1 [ReduceSum]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Gather_3_output_0
[08/01/2023-08:17:33] [V] [TRT] /ReduceSum_1 [ReduceSum] inputs: [/Gather_3_output_0 -> (1200)[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /ReduceSum_1 for ONNX node: /ReduceSum_1
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /ReduceSum_1_output_0 for ONNX tensor: /ReduceSum_1_output_0
[08/01/2023-08:17:33] [V] [TRT] /ReduceSum_1 [ReduceSum] outputs: [/ReduceSum_1_output_0 -> ()[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Cast_1 [Cast]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /ReduceSum_1_output_0
[08/01/2023-08:17:33] [V] [TRT] /Cast_1 [Cast] inputs: [/ReduceSum_1_output_0 -> ()[FLOAT]], 
[08/01/2023-08:17:33] [V] [TRT] Casting to type: int32
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /Cast_1 for ONNX node: /Cast_1
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /Cast_1_output_0 for ONNX tensor: /Cast_1_output_0
[08/01/2023-08:17:33] [V] [TRT] /Cast_1 [Cast] outputs: [/Cast_1_output_0 -> ()[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Constant_10 [Constant]
[08/01/2023-08:17:33] [V] [TRT] /Constant_10 [Constant] inputs: 
[08/01/2023-08:17:33] [V] [TRT] /Constant_10 [Constant] outputs: [/Constant_10_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Constant_11 [Constant]
[08/01/2023-08:17:33] [V] [TRT] /Constant_11 [Constant] inputs: 
[08/01/2023-08:17:33] [V] [TRT] /Constant_11 [Constant] outputs: [/Constant_11_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Unsqueeze [Unsqueeze]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Cast_output_0
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Constant_11_output_0
[08/01/2023-08:17:33] [V] [TRT] /Unsqueeze [Unsqueeze] inputs: [/Cast_output_0 -> ()[INT32]], [/Constant_11_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Original shape: (), unsqueezing to: (1,)
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /Unsqueeze for ONNX node: /Unsqueeze
[08/01/2023-08:17:33] [V] [TRT] Registering tensor: /Unsqueeze_output_0 for ONNX tensor: /Unsqueeze_output_0
[08/01/2023-08:17:33] [V] [TRT] /Unsqueeze [Unsqueeze] outputs: [/Unsqueeze_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Constant_12 [Constant]
[08/01/2023-08:17:33] [V] [TRT] /Constant_12 [Constant] inputs: 
[08/01/2023-08:17:33] [V] [TRT] /Constant_12 [Constant] outputs: [/Constant_12_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Constant_13 [Constant]
[08/01/2023-08:17:33] [V] [TRT] /Constant_13 [Constant] inputs: 
[08/01/2023-08:17:33] [V] [TRT] /Constant_13 [Constant] outputs: [/Constant_13_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Parsing node: /Slice_2 [Slice]
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Squeeze_output_0
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Constant_10_output_0
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Unsqueeze_output_0
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Constant_12_output_0
[08/01/2023-08:17:33] [V] [TRT] Searching for input: /Constant_13_output_0
[08/01/2023-08:17:33] [V] [TRT] /Slice_2 [Slice] inputs: [/Squeeze_output_0 -> (1200, 1200)[FLOAT]], [/Constant_10_output_0 -> (1)[INT32]], [/Unsqueeze_output_0 -> (1)[INT32]], [/Constant_12_output_0 -> (1)[INT32]], [/Constant_13_output_0 -> (1)[INT32]], 
[08/01/2023-08:17:33] [V] [TRT] Registering layer: /Slice_2 for ONNX node: /Slice_2
[08/01/2023-08:17:33] [E] Error[4]: [graphShapeAnalyzer.cpp::demandAndResolveInputTensor::1247] Error Code 4: Internal Error (onnx::Concat_2: network input that is shape tensor must have type Int32)
[08/01/2023-08:17:33] [E] [TRT] ModelImporter.cpp:726: While parsing node number 362 [Slice -> "/Slice_2_output_0"]:
[08/01/2023-08:17:33] [E] [TRT] ModelImporter.cpp:727: --- Begin node ---
[08/01/2023-08:17:33] [E] [TRT] ModelImporter.cpp:728: input: "/Squeeze_output_0"
input: "/Constant_10_output_0"
input: "/Unsqueeze_output_0"
input: "/Constant_12_output_0"
input: "/Constant_13_output_0"
output: "/Slice_2_output_0"
name: "/Slice_2"
op_type: "Slice"

[08/01/2023-08:17:33] [E] [TRT] ModelImporter.cpp:729: --- End node ---
[08/01/2023-08:17:33] [E] [TRT] ModelImporter.cpp:732: ERROR: ModelImporter.cpp:185 In function parseGraph:
[6] Invalid Node - /Slice_2
[graphShapeAnalyzer.cpp::demandAndResolveInputTensor::1247] Error Code 4: Internal Error (onnx::Concat_2: network input that is shape tensor must have type Int32)
[08/01/2023-08:17:33] [E] Failed to parse onnx file
[08/01/2023-08:17:33] [I] Finish parsing network model
[08/01/2023-08:17:33] [E] Parsing model failed
[08/01/2023-08:17:33] [E] Failed to create engine from model or file.
[08/01/2023-08:17:33] [E] Engine set up failed`

**Commands or scripts**:
torch to onnx script：

    import torch
    from torch.utils.data import Dataset, DataLoader
    from exps.exp import Exp
    import numpy as np
    
    exp = Exp()
    model = exp.get_model()
    
    model_file = "./transvcl/weights/model.pth"
    ckpt = torch.load(model_file)
    model.load_state_dict(ckpt['model'])
    model.to("cpu")
    model.eval()
    
    """
    Args:
        feat0 (torch.Tensor): [N, L, C]
        feat1 (torch.Tensor): [N, S, C]
        mask0 (torch.Tensor): [N, L] (optional)
        mask1 (torch.Tensor): [N, S] (optional)
    """            
    dummy_input = (torch.randn(1, 1200, 256, requires_grad=True),
                   torch.randn(1, 1200, 256, requires_grad=True),
                   torch.randn(1, 1200, requires_grad=True),
                   torch.randn(1, 1200, requires_grad=True),
                   ["abc"]
                   )
    
    torch.onnx.export(model,
                    dummy_input,
                    "./weights/transvcl.onnx",
                    export_params=True,
                    opset_version=13,
                    do_constant_folding=True,
                    input_names=['input'],
                    output_names=['output'])

onnx to trt exec：
    /usr/src/tensorrt/bin/trtexec --onnx=./weights/transvcl.onnx --saveEngine=./weights/transvcl_sim.trt

![image](https://github.com/NVIDIA/TensorRT/assets/101395844/0e13c5e9-ab79-44e0-9a8f-9f7144b4fee8)

**Have you tried [the latest release](https://developer.nvidia.com/tensorrt)?**:

**Can this model run on other frameworks?** For example run ONNX model with ONNXRuntime (`polygraphy run <model.onnx> --onnxrt`):

**Comment by zerollzeng** [comment]
> [6] Invalid Node - /Slice_2
> [graphShapeAnalyzer.cpp::demandAndResolveInputTensor::1247] Error Code 4: Internal Error (onnx::Concat_2: network input that is shape tensor must have type Int32)

Check this node, the shape tensor input should have datatype int32, I guess it's FP32 in your model, try re-export it or do constant folding with polygraphy.

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks according to our policy. Thanks!
